src/PostgREST/Types.hs

```diff
@@ -157,13 +159,17 @@ type ProcsMap = M.HashMap QualifiedIdentifier [ProcDescription]
   An overloaded function can have a different volatility or even a different return type.
   Ideally, handling overloaded functions should be left to pg itself. But we need to know certain proc attributes in advance.
 -}
-findProc :: QualifiedIdentifier -> S.Set Text -> Bool -> ProcsMap -> Maybe ProcDescription
-findProc qi payloadKeys paramsAsSingleObject allProcs =
-  case M.lookup qi allProcs of
-    Nothing     -> Nothing
-    Just [proc] -> Just proc           -- if it's not an overloaded function then immediately get the ProcDescription
-    Just procs  -> find matches procs  -- Handle overloaded functions case
+findProc :: QualifiedIdentifier -> S.Set Text -> Bool -> ProcsMap -> ProcDescription
+findProc qi payloadKeys paramsAsSingleObject allProcs = fromMaybe fallback bestMatch
   where
+    -- instead of passing Maybe ProcDescription around, we create a fallback description here when we can't find a matching function
+    -- args is empty, but because "specifiedProcArgs" will fill the missing arguments with default type text, this is not a problem
+    fallback = ProcDescription (qiSchema qi) (qiName qi) (Just "fallback descriptor") mempty (SetOf $ Composite $ QualifiedIdentifier "" "record") Volatile
```

Comment:
Hmm... how about making it like:
```suggestion
    fallback = ProcDescription (qiSchema qi) (qiName qi) Nothing mempty (SetOf $ Composite $ QualifiedIdentifier "" "record") Volatile
```
The  `(SetOf $ Composite $ QualifiedIdentifier "" "record")` doesn't look right though.
To be more correct, shouldn't we default to returning `void`?

Reply:
> `SetOf $ Single $`

That doesn't make sense. Did you want to make it `SetOf Scalar` or `Single Composite`?

In any case: We need to default to `SetOf Composite` to at least return all results in any kind of format in this case.

Reply:
> "unknown" reads quite well in the list as well, but I understand that type to be associated with scalars of unknown type and not.. records.

Agree. record is good. Just changing the description to Nothing would be fine.

> SetOf $ Single $
> That doesn't make sense.

Yeah, sorry, I messed up there. Disregard that suggestion.